docs/enhancements/crypto-api.md

```diff
@@ -0,0 +1,246 @@
+    smbios:
+      <redacted>
+  customizeComponents: {}
+  tlsSecurityProfile:
+    type: Intermediate
+    intermediate: {}
+  uninstallStrategy: BlockUninstallIfWorkloadsExist
+  workloadUpdateStrategy:
+    batchEvictionInterval: 1m0s
+    batchEvictionSize: 10
+    workloadUpdateMethods:
+    - LiveMigrate
+```
+
+
+HCO will then be reconciling this value for all underlying HCO managed CRs, in their respective stanza.
+
+The custom resources that the TLS Security Profile stanza will be propagated to, are:
+* `KubeVirt kubevirt.io/v1 - kubevirt.kubevirt.io/kubevirt-kubevirt-hyperconverged`
+* `CDI cdi.kubevirt.io/v1beta1 - cdis.cdi.kubevirt.io/cdi-kubevirt-hyperconverged`
+* `NetworkAddonsConfig networkaddonsoperator.network.kubevirt.io/v1 - networkaddonsconfigs.networkaddonsoperator.network.kubevirt.io/cluster`
+* `SSP ssp.kubevirt.io/v1beta1 - ssps.ssp.kubevirt.io/ssp-kubevirt-hyperconverged`
+
+
+## Notes
+
+1. Any modification for the TLS Security Profile for HCO managed components must be made through APIServer CR and they can happen at any time (day-2 configurations).
+2. Modification of TLS Security Profile for a specific component will be possible by using a [jsonpatch annotation](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/docs/cluster-configuration.md#jsonpatch-annotations) on HCO CR. However, this configuration is unsupported and should block upgrades.
+3. This feature/configuration is only applicable to OCP/OKD platform, not plain kubernetes, since APIServer (apiversion: config.openshift.io/v1) is an openshift resource only.
+4. [https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations](https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations) also mentions Certificate type and Certificate lifespan but [https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/tls-config.md](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/tls-config.md) doesn't, so we can safely ignore them
+
+
+## Open Questions
+1. In plain kubernetes HCO deployment, will we use the default intermediate profile for all HCO managed components, or give the option for the administrator to set it on the HCO CR, which will propagate to all components?
+   1. The TLS Security Profile configuration feature will be available only on OCP/OKD platforms.
+   2. We can decide to expose the API to let the user optionally fine-tune/override the configuration for all the Hyperconverged Kubevirt having HCO reading the OpenShift API server one as its default only if the user didn't express custom values. This will work also on plain k8s.
+
+2. [OpenShift API design](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/tls-config.md) explicitly mentions API servers, but it never mentions admission webhooks that are also TLS protected.
+Is there any technical reason for us to ignore the admission webhooks? 
```

Comment:
I sincerely believe those settings are not expected to be propagated down to services that are not directly accessible from outside the cluster like our webhooks/aggregated apiservers.